ggml-qnn.cpp

```diff
@@ -0,0 +1,3196 @@
+        powerConfig.dcvsV3Config.busVoltageCornerMin = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        powerConfig.dcvsV3Config.busVoltageCornerTarget = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        powerConfig.dcvsV3Config.busVoltageCornerMax = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        // set Core Clock Parameters (refer QnnHtpPerfInfrastructure_VoltageCorner_t enum)
+        powerConfig.dcvsV3Config.coreVoltageCornerMin = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        powerConfig.dcvsV3Config.coreVoltageCornerTarget = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        powerConfig.dcvsV3Config.coreVoltageCornerMax = DCVS_VOLTAGE_VCORNER_MAX_VOLTAGE_CORNER;
+        // set power config with different performance parameters
+        const QnnHtpPerfInfrastructure_PowerConfig_t *powerConfigs[] = {&powerConfig, nullptr};
+
+        _qnn_htp_perfinfra->setPowerConfig(_qnn_power_configid, powerConfigs);
+
+        return 0;
+    }
+
+    std::string & get_qnn_graph_name() { return _graph_name; }
+
+    bool is_rpcmem_initialized() {
+        return _rpcmem_initialized;
+    }
+
+    void set_rpcmem_initialized(bool initialized) {
+        _rpcmem_initialized = initialized;
+    }
+
+    int32_t rpcmem_to_fd(void * buf);
+
+    int register_rpcmem(void * p_data, Qnn_Tensor_t * p_tensor);
+
+    void unregister_rpcmem();
+
+    void *alloc_rpcmem(size_t bytes, size_t alignment);
+
+    void free_rpcmem(void * buf);
+
+    bool is_rpcmem_allocated(void * buf);
+
+    bool is_rpcmem_registered(Qnn_MemHandle_t handle) {
+        return _qnn_mem_set.count(handle) != 0U;
+    }
+
+public:
+    std::map<std::string, std::tuple<Qnn_GraphHandle_t, Qnn_Tensor_t *, Qnn_Tensor_t *, Qnn_Tensor_t *>> _qnn_graph_map;
+
+private:
+    int load_system();
+
+    int unload_system();
+
+    int load_backend(std::string &lib_path, const QnnSaver_Config_t ** saver_config);
+
+    int unload_backend();
+
+    void set_qnn_raw_interface(QNN_INTERFACE_VER_TYPE & raw_interface) {
+        _qnn_raw_interface = raw_interface;
+    }
+
+    void set_qnn_raw_system_interface(QNN_SYSTEM_INTERFACE_VER_TYPE &raw_interface) {
+        _qnn_raw_system_interface = raw_interface;
+    }
+
+private:
+    static constexpr const int _required_num_providers = 1;
+
+private:
+    std::string _lib_path;
+    std::string _backend_name;
+    std::string _model_name;                         // prebuilt QNN model name, not used in currently
+    BackendIdType _backend_id;
+
+    bool _debug_tensor                      = false; // flag to indicate if requested graph is to be run in debug mode
+    bool _do_node_validations               = true;  // flag to indicate whether all add_node calls need to be validated
+    QnnLog_Level_t _qnn_log_level           = QNN_LOG_LEVEL_DEBUG;
+
+    ggml_qnn_profile_level _profile_level   = ggml_qnn_profile_level::profile_detail;
+
+    qnn_interface _qnn_interface;
+
+    void * _system_lib_handle = nullptr;
```

Comment:
nit: different format rule vs line 907

Reply:
what's the meaning of this?

Reply:
might be it's not important here.

Reply:
consider aligning the identifiers before the `=` for improved readability. 
utilizing source code formatting tools, such as `clang-format`, can automate this process and ensure consistent code style throughout the project.

Reply:
Thanks. I see. it's about text format rule.I will modify it accordingly.